stdlib/json/__init__.pyi

```diff
@@ -33,6 +33,19 @@ def dump(
     sort_keys: bool = ...,
     **kwds: Any,
 ) -> None: ...
+@overload
+def loads(
+    s: str | bytes,
+    *,
+    cls: Type[JSONDecoder] | None = ...,
+    object_hook: None = ...,
+    parse_float: None = ...,
+    parse_int: None = ...,
+    parse_constant: None = ...,
+    object_pairs_hook: None = ...,
+    **kwds: Any,
+) -> Json: ...
```

Comment:
I don't like how this introduces an implicit union return and breaks code like `json.loads(some_string)["Foo"]` breaks. It is more correct, but the usual reasons for avoiding union returns apply here IMO.

Reply:
True, another good reason for python/typing#566.